compiler/fir/raw-fir/psi2fir/src/org/jetbrains/kotlin/fir/builder/RawFirBuilder.kt

```diff
@@ -1078,19 +1086,24 @@ class RawFirBuilder(
                                 expression = { delegateExpression }.toFirExpression("Should have delegate")
                             }
                         } else null
+
+                        getter = this@toFirProperty.getter.toFirPropertyAccessor(this@toFirProperty, propertyType, isGetter = true)
+                        setter = if (isVar) {
+                            this@toFirProperty.setter.toFirPropertyAccessor(this@toFirProperty, propertyType, isGetter = false)
+                        } else null
+
                         status = FirDeclarationStatusImpl(visibility, modality).apply {
                             isExpect = hasExpectModifier()
                             isActual = hasActualModifier()
                             isOverride = hasModifier(OVERRIDE_KEYWORD)
                             isConst = hasModifier(CONST_KEYWORD)
                             isLateInit = hasModifier(LATEINIT_KEYWORD)
+                            isInline = hasModifier(INLINE_KEYWORD) ||
+                                    (!isVar && getter!!.isInline) || (isVar && getter!!.isInline && setter!!.isInline)
```

Comment:
IMO this is over-complicated. Something like `getter!!.isInline && setter?.isInline == true` looks much simpler. Same thing for code below.

Reply:
Agree, but I've tried to be as explicit as possible. :)

I worried `setter?.isInline == true` can't catch my intention, though. It implies setter is not null _and_ the specific modifier exists. However, for `val`, setter doesn't exist and it should trigger the upward propagation as well. Hence, `setter?.isInline != false` seems right to me. I added some comments to explain that as well.